javascript/javascript3/week3/shopping/shopping.js

```diff
@@ -0,0 +1,126 @@
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  getCurrency(currency) {
+    switch (currency) {
+      case "GBP":
+        return "£" + this.price * 0.11;
+
+      case "USD":
+        return this.price * 0.15 + "USD";
+
+      case "INR":
+        return this.price * 11.24 + " INR ";
+
+      case "euros":
+        return this.price * 0.13 + " EUR";
+
+      default:
+        return this.price + " DKK";
+    }
+  }
+}
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    // Implement functionality here
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    // Implement functionality here
+    for (let i = 0; i < this.products.length; i++) {
+      if (product === this.products[i]) {
+        this.products.splice(i, 1);
+      }
+    }
+  }
+
+  searchProduct(productName) {
+    // Implement functionality here
+    let sProduct = [];
+    this.products.forEach((product) => {
+      if (product.name.toLowerCase() === productName.toLowerCase()) {
+        sProduct = product;
+      }
+    });
+    return sProduct;
+  }
+
+  getTotal() {
+    // Implement functionality here
+    const totalPrice = this.products.reduce((sum, product) => {
+      return (sum += product.price);
+    }, 0);
+    return totalPrice;
+  }
+
+  renderProducts() {
+    // Implement functionality here
+    const productsList = document.getElementById("productsList");
+    this.products.forEach((product) => {
+      const productItems = document.createElement("li");
+      productItems.innerHTML = `<h2>${product.name}</h2> <span>${product.price}</span>`;
+      productsList.appendChild(productItems);
+    });
+  }
+
+  getUser(user) {
+    // Implement functionality here
+    return new Promise((resolve, reject) => {
+      fetch(`https://jsonplaceholder.typicode.com/users/${user}`)
+        .then((response) => response.json())
+        .then((data) => resolve(data));
```

Comment:
I prefer to use resolve before the fetch call.

Reply:
fixed :)